File: app/src/main/java/com/tytech/ninjawarrior1337/supremeapp/MainActivity.java
package com.tytech.ninjawarrior1337.supremeapp;

import android.content.Context;
import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.ContextMenu;
import android.view.Menu;
import android.view.MenuItem;
import android.view.MotionEvent;
import android.view.View;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.Toast;

import io.kimo.konamicode.KonamiCode;
import io.kimo.konamicode.KonamiCodeLayout;

import static android.R.attr.bottom;
import static android.R.attr.content;
import static android.R.attr.fragment;
import static android.R.attr.top;
import static android.R.attr.visibility;
import static android.R.attr.visible;
import static android.os.Build.VERSION_CODES.M;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        new KonamiCode.Installer(MainActivity.this).on(this).callback(new KonamiCodeLayout.Callback()
        {
            @Override
            public void onFinish() {
                Toast.makeText(MainActivity.this, "Vert Dnag", Toast.LENGTH_SHORT).show();
            }
        }).install();

        final Button botomPortal;
        botomPortal = (Button) findViewById(R.id.bottomPortal);

        final Button topPortal;
        topPortal = (Button) findViewById(R.id.topPortal);

        final ImageView finegear;
        finegear = (ImageView) findViewById(R.id.finegear);
        finegear.setVisibility(View.INVISIBLE);

        botomPortal.setOnTouchListener(new View.OnTouchListener()
        {
            @Override
            public boolean onTouch(View v, MotionEvent event)
            {
                switch (event.getAction()){
                    case MotionEvent.ACTION_DOWN:
                        finegear.setVisibility(View.VISIBLE);
                        break;
                    case MotionEvent.ACTION_UP:
                        finegear.setVisibility(View.INVISIBLE);
                        break;
                }
                return false;
            }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.overflow, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item)
    {
        if(item.getItemId() == R.id.settings)
        {
            Intent myIntent = new Intent(MainActivity.this, SettingsActivity.class);
            startActivity(myIntent);
        }
        return true;
    }
}

File: app/src/main/java/com/tytech/ninjawarrior1337/supremeapp/MainActivity.kt
package com.tytech.ninjawarrior1337.supremeapp

import android.annotation.SuppressLint
import android.content.Intent
import android.graphics.drawable.Drawable
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Button
import android.widget.ImageView
import android.widget.Toast
import com.larvalabs.svgandroid.SVGParser
import io.kimo.konamicode.KonamiCode
import android.R.raw
import android.app.Activity
import android.app.PendingIntent.getActivity
import android.content.Context
import android.content.SharedPreferences
import android.content.res.ColorStateList
import android.content.res.Resources
import android.graphics.Color
import android.support.v4.content.ContextCompat
import android.support.v4.content.ContextCompat.startActivity
import android.support.v4.view.ViewCompat.setBackgroundTintList
import android.util.Log
import android.view.*
import com.larvalabs.svgandroid.SVG
import com.tytech.ninjawarrior1337.R

class MainActivity : AppCompatActivity() {
    @SuppressLint("ClickableViewAccessibility", "WrongViewCast")
    override fun onCreate(savedInstanceState: Bundle?)
    {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        KonamiCode.Installer(this@MainActivity).on(this).callback { Toast.makeText(this@MainActivity, "Vert Dnag", Toast.LENGTH_SHORT).show() }.install()

        val botomPortalB: Button
        botomPortalB = findViewById(R.id.bottomPortal) as Button

        val topPortalB: Button
        topPortalB = findViewById(R.id.topPortal) as Button

        val PREFS_NAME = "SupremeAppPrefs"
        var pref = this.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)

        pref.edit().putString("topPortalC", "#" + Integer.toHexString(R.color.defaultTPortal)).apply()
        pref.edit().putString("bottomPortalC", "#" + Integer.toHexString(R.color.defaultBPortal)).apply()


        val finegear: ImageView
        finegear = findViewById(R.id.finegear) as ImageView
        finegear.visibility = View.INVISIBLE

        botomPortalB.setOnTouchListener { v, event ->
            when (event.action)
            {
                MotionEvent.ACTION_DOWN -> findViewById(R.id.finegear).visibility = View.VISIBLE
                MotionEvent.ACTION_UP -> findViewById(R.id.finegear).visibility = View.INVISIBLE
            }
            false
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.overflow, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.settings) {
            val myIntent = Intent(this@MainActivity, SettingsActivity::class.java)
            startActivity(myIntent)
        }
        return true
    }

//    public override fun onResume()
//    {
//        super.onResume()
//        val PREFS_NAME = "SupremeAppPrefs"
//        var pref = this.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
//        setBackgroundTintList(findViewById(R.id.topPortal), ColorStateList.valueOf(Color.parseColor(pref.getString("topPortalC", Integer.toHexString(R.color.defaultTPortal)))))
//        setBackgroundTintList(findViewById(R.id.bottomPortal), ColorStateList.valueOf(Color.parseColor(pref.getString("bottomPortalC", Integer.toHexString(R.color.defaultBPortal)))))
//
//    }
    public override fun onRestart()
    {
        super.onRestart()

        val PREFS_NAME = "SupremeAppPrefs"
        var pref = this.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
        setBackgroundTintList(findViewById(R.id.topPortal), ColorStateList.valueOf(Color.parseColor(pref.getString("topPortalC", Integer.toHexString(R.color.defaultTPortal)))))
        setBackgroundTintList(findViewById(R.id.bottomPortal), ColorStateList.valueOf(Color.parseColor(pref.getString("bottomPortalC", Integer.toHexString(R.color.defaultBPortal)))))

        //Log.d("vertd", pref.getInt("topPortalC", R.color.defaultTPortal).toString())

    }

    override fun onStop() {
        super.onStop()
        val PREFS_NAME = "SupremeAppPrefs"
        var pref = this.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
    }
}
